### PR TITLE
7 endpoint

### PR DESCRIPTION
**Pull request by yainporter**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [ ] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s): 
EP - 7, happy path only
*Brief summary of the change(s) included in this pull request*
- Removed the update action in Vendors
- Add association to Vendors
- Add route for delete
- Add controller#destroy
- Add testing for end point

## Implements/Fixes:

description closes #

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [x] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*

## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

### CODE REVIEW

File: app/models/vendor.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 class Vendor < ApplicationRecord
-
-end
\ No newline at end of file
+has_many :market_vendors, dependent: :destroy
+end
```
Comment (odellmac4) [inline app/models/vendor.rb:3]: Will we be using this association in another endpoint as I'm not seeing it used in the code.

Do we need to create/test all associations?

Comment (yainporter) [reply app/models/vendor.rb:3]: The association needs to be set up so that the market_vendor can be destroyed as well when a vendor is destroyed

Comment (yainporter) [reply app/models/vendor.rb:3]: I think any association that we create should be tested, I can add the edge case testing to make it more robust

Comment (odellmac4) [reply app/models/vendor.rb:3]: Got it!

Please confirm once you've updated your tests and I can review and merge

File: spec/requests/api/v0/vendors_request_spec.rb [hunk 2]
```diff
@@ -26,4 +26,51 @@
...
+    expect(response).to be_successful
+    expect(response.code).to eq("204")
+    expect(response).to have_http_status(:no_content)
+    expect(Vendor.count).to eq(0)
+    expect{Vendor.find(vendor.id)}.to raise_error(ActiveRecord::RecordNotFound)
+    expect(MarketVendor.count).to eq(0)
+    expect{MarketVendor.find(market_vendor.id)}.to raise_error(ActiveRecord::RecordNotFound)
+    expect(Market.count).to eq(1)
+    expect(Market.find(market.id)).to eq(market)
+  end
```
Comment (odellmac4) [inline spec/requests/api/v0/vendors_request_spec.rb:75]: Testing looks good and thanks for explaining that concept to me!

**Review by odellmac4 — APPROVED** [review]
